src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import core.basesyntax.service.ReaderService;
+import core.basesyntax.service.ReportService;
+import core.basesyntax.service.StorageService;
+import core.basesyntax.service.WriterService;
+import core.basesyntax.service.impl.ReaderServiceImpl;
+import core.basesyntax.service.impl.ReportsServiceImpl;
+import core.basesyntax.service.impl.StorageServiceImpl;
+import core.basesyntax.service.impl.WriteServiceImpl;
+import java.util.List;
+
+public class Main {
+    public static void main(String[] args) {
+        final String inputFile = "src/main/resources/database.csv";
+        final String outputFile = "src/main/resources/result.csv";
+
+        ReaderService readerService = new ReaderServiceImpl();
+        StorageService storageService = new StorageServiceImpl();
+        ReportService reportService = new ReportsServiceImpl();
+        WriterService writerService = new WriteServiceImpl();
+        List<String> dateFromFile = readerService.readFromFile(inputFile);
+        storageService.fillStorage(dateFromFile);
```

Comment:
You have parsing inside fillStorage(). Parsing is independent operation, create class TransactionParser for it. Remember SRP.
```suggestion
                TransactionParser parser = new TransactionParser();
                List<FruitTransaction> transactions = parser.parse(dateFromFile);
                TransactionExecutor executor = new TransactionExecutorImpl(operationStrategy);
                executor.executeTransactions(transactions);
```